app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -25,11 +25,12 @@ def prov_field_changed
       refresh_divs = prov_get_form_vars # Get changed option, returns true if divs need refreshing
       build_grid if refresh_divs
       changed = (@edit[:new] != @edit[:current])
-      @record = if @edit[:new][:src_configured_system_ids].present?
-                  PhysicalServer.find(@edit[:new][:src_configured_system_ids].first)
-                elsif @edit[:new][:src_vm_id].first.present?
-                  MiqTemplate.find(@edit[:new][:src_vm_id].first)
-                end
+      @record =
+        if @edit[:new][:src_configured_system_ids].present?
+          PhysicalServer.where(:id => @edit[:new][:src_configured_system_ids].first).first
```

Comment:
Maybe [`take`](https://apidock.com/rails/ActiveRecord/FinderMethods/take) is semantically more accurate instead of `first`, but I leave the decision to you.